src/prefect/results.py

```diff
@@ -504,7 +492,12 @@ async def create(
         key = uuid.uuid4().hex
         await storage_block.write_path(key, content=blob.to_bytes())
 
-        description = cls._infer_description(obj, storage_block, key)
+        description = f"Result of type {type(obj).__name__!r}"
+        uri = cls._infer_path(storage_block, key)
+        if uri:
+            description += f" persisted to {uri}."
```

Comment:
```suggestion
            description += f" Persisted to {uri}."
```

Reply:
this gets appended to the previous string so we don't want to capitalize it